src/pages/ReimbursementAccount/CompanyStep.js

```diff
@@ -91,56 +123,74 @@ class CompanyStep extends React.Component {
     }
 
+    /**
+     * Clear the error associated to inputKey if found and store the inputKey new value in the state.
+     *
+     * @param {String} inputKey
+     * @param {String} value
+     */
+    clearErrorAndSetValue(inputKey, value) {
+        this.setValue({[inputKey]: value});
+        const errors = this.getErrors();
+        if (!errors[inputKey]) {
+            // No error found for this inputKey
+            return;
+        }
+
+        // Clear the existing error for this inputKey
+        const newErrors = {...errors};
+        delete newErrors[inputKey];
+        setBankAccountFormValidationErrors(newErrors);
+    }
+
     /**
      * @returns {Boolean}
      */
     validate() {
+        const errors = {};
         if (!this.state.password.trim()) {
-            showBankAccountFormValidationError(this.props.translate('common.passwordCannotBeBlank'));
-            return false;
+            errors.password = true;
         }
 
         if (!isValidAddress(this.state.addressStreet)) {
-            showBankAccountFormValidationError(this.props.translate('bankAccount.error.addressStreet'));
-            return false;
+            errors.addressStreet = true;
         }
 
         if (this.state.addressState === '') {
-            showBankAccountFormValidationError(this.props.translate('bankAccount.error.addressState'));
-            return false;
+            errors.addressState = true;
         }
 
         if (!isValidZipCode(this.state.addressZipCode)) {
-            showBankAccountFormValidationError(this.props.translate('bankAccount.error.zipCode'));
-            return false;
+            errors.addressZipCode = true;
         }
 
         if (!Str.isValidURL(this.state.website)) {
-            showBankAccountFormValidationError(this.props.translate('bankAccount.error.website'));
-            return false;
+            errors.website = true;
         }
 
         if (!/[0-9]{9}/.test(this.state.companyTaxID)) {
-            showBankAccountFormValidationError(this.props.translate('bankAccount.error.taxID'));
-            return false;
+            errors.companyTaxID = true;
         }
 
         if (!isValidDate(this.state.incorporationDate)) {
-            showBankAccountFormValidationError(this.props.translate('bankAccount.error.incorporationDate'));
-            return false;
+            errors.incorporationDate = true;
         }
 
         if (!isValidIndustryCode(this.state.industryCode)) {
-            showBankAccountFormValidationError(this.props.translate('bankAccount.error.industryCode'));
-            return false;
+            errors.industryCode = true;
         }
 
         if (!this.state.hasNoConnectionToCannabis) {
-            showBankAccountFormValidationError(this.props.translate('bankAccount.error.restrictedBusiness'));
-            return false;
+            errors.hasNoConnectionToCannabis = true;
         }
+        this.requiredFields.forEach((inputKey) => {
```

Comment:
`_.each()`

Reply:
I see that `_.each` is more common than `.forEach`, and also it seems we prefer using `_.map` rather than `.map`. I understand from the point of making the code look the same to stick to what we have been doing.... but I wonder why are we sticking to use `underscore` instead of using new javascript features (not so new anymore).

Things I like about `.forEach`, `.map`, `.filter`:

- They are part of javascript, no need to use a library. From the last years developing in javascript, I have much fresher the new features from ES6 than `underscore`, which I used to use in the past.
- Mentioned array functions are easier to chain and, in my opinion, the way the expression is laid out looks more readable. (i.e. `someArray.forEach(dosomething)` looks nicer than `_.each(someArray, dosomething)`). Chaining in `underscore` has to be done by adding `_.chain` and `_.value`.

One subtle difference between the two explained [here](https://stackoverflow.com/a/18884620/16434681), but I wouldn't expect this is something we are doing here.

The advantage I can think of using `underscore` methods is that they are more ""relaxed"" and may handle more things than just arrays (i.e. you can iterate an object).

I think this last "advantage" is something that adds flexibility to our code, but it also makes it less obvious about what types we are handling. 

If I was able to choose, I would prefer to move toward relying less in `underscore` which now days is less necessary with new javascript features. Write more boiler plate when we iterate objects using code like `Object.keys(someobject).map`, and use the standard available functions in arrays like `map`, `forEach`, `filter`, etc.

Well, that's all about that subject, if we still think we are better off using `underscore` functions, I'll stick to them! 😬